students/aravichander/lesson09/Mailroom_class.py

```diff
@@ -3,68 +3,105 @@
 import sys
 import json
 import unittest 
+import math 
 
-class Donor:
-	def __init__(self,Name):
-		self.Name = Name
-
-	@property
-	def donations(self):
-		return self.donations
-
-	@classmethod
-	def sumdonation(cls, donations):
-		total = sum(self.donations)
-		return cls.Donor(Name)
-
-class Donorlist:
-	def __init__(self,Donors)
 
+class Donor:
+	def __init__(self,name,donations = None): #add default parameters for donations
+		self._name = name
+		self._donations = [] if donations is None else donations 
 
+	def add_donation(self,donation):
+		self._donations.append(donation)
 
 	@property
-	def numberofdonors(self)
+	def name(self):
+		return self._name
 
 	@property
-	def totaldonationsofdonors(self)
-
-	@classmethod
-	def memberdonations(cls..etc.)
-
-
-
-
-
-
-
-
-
-
-
-
-Donors = {}
-
-def createlist():
-	Donors["Bill Gates"] = [10000,12000]
-	Donors["Jeff Bezos"]=[50]
-	Donors["Mark Zuckerberg"] = [500,600,700]
-	Donors["Paul Allen"] = [250,350,450]
-	Donors["King of Siam"] = [200,250]
-
-def getlist():
-	return Donors 
-	
-
-def printdonors():
-	for x,v in Donors.items():
-		print("\n{} donated ${}".format(x,v))
+	def donations(self):
+		return self._donations
 
-def thankall():
-	for x in Donors:
-		totaldonated = sum(Donors[x])
-		letterofthanks(x,totaldonated)
+	@property
+	def sum_donation(self):
+		return sum(self._donations)
 
-def letterofthanks(donorname,totaldonated):
+	@property
+	def number_of_donations(self):
+		return len(self._donations)
+#My ipython was going nuts so I had to manually calculate the average >=(
+	@property
+	def donation_average(self):
+		return sum(self._donations)/len(self._donations)
+
+
+
+class Donor_db:
+	def __init__(self):
+		self._donors = {}
+
+	def add_donor(self,donor):
+		self._donors[donor.name.lower()] = donor
+
+	# def all_donors_donations(self): Couldn't figure out the issue here re:testing
```

Comment:
hmm -- looks OK to me.

though it could be more compact:

`total_sum = sum((d.sum_donation for d in self._donors.items()))`

or something like that -- untested